extensions/standard-processors/tests/unit/AttributesToJSONTests.cpp

```diff
@@ -0,0 +1,194 @@
+ */
+#include <string>
+#include <vector>
+
+#include "rapidjson/writer.h"
+#include "rapidjson/stringbuffer.h"
+#include "TestBase.h"
+#include "utils/TestUtils.h"
+#include "AttributesToJSON.h"
+#include "GetFile.h"
+#include "PutFile.h"
+#include "UpdateAttribute.h"
+#include "LogAttribute.h"
+
+namespace {
+
+class AttributesToJSONTestFixture {
+ public:
+  const std::string TEST_FILE_CONTENT = "test_content";
+  const std::string TEST_FILE_NAME = "tstFile.ext";
+
+  AttributesToJSONTestFixture() {
+    LogTestController::getInstance().setTrace<TestPlan>();
+    LogTestController::getInstance().setDebug<minifi::processors::AttributesToJSON>();
+    LogTestController::getInstance().setDebug<minifi::processors::GetFile>();
+    LogTestController::getInstance().setDebug<minifi::processors::PutFile>();
+    LogTestController::getInstance().setDebug<minifi::processors::UpdateAttribute>();
+    LogTestController::getInstance().setDebug<minifi::processors::LogAttribute>();
+
+    dir_ = test_controller_.createTempDirectory();
+
+    plan_ = test_controller_.createPlan();
+    getfile_ = plan_->addProcessor("GetFile", "GetFile");
+    update_attribute_ = plan_->addProcessor("UpdateAttribute", "UpdateAttribute", core::Relationship("success", "description"), true);
+    attribute_to_json_ = plan_->addProcessor("AttributesToJSON", "AttributesToJSON", core::Relationship("success", "description"), true);
+    logattribute_ = plan_->addProcessor("LogAttribute", "LogAttribute", core::Relationship("success", "description"), true);
+    putfile_ = plan_->addProcessor("PutFile", "PutFile", core::Relationship("success", "description"), true);
+
+    plan_->setProperty(getfile_, org::apache::nifi::minifi::processors::GetFile::Directory.getName(), dir_);
+    plan_->setProperty(putfile_, org::apache::nifi::minifi::processors::PutFile::Directory.getName(), dir_);
+
+    update_attribute_->setDynamicProperty("my_attribute", "my_value");
+    update_attribute_->setDynamicProperty("other_attribute", "other_value");
+    update_attribute_->setDynamicProperty("empty_attribute", "");
+
+    std::fstream file;
+    std::stringstream ss;
+    ss << dir_ << utils::file::FileUtils::get_separator() << TEST_FILE_NAME;
+    file.open(ss.str(), std::ios::out);
+    file << TEST_FILE_CONTENT;
+    file.close();
+  }
+
+  std::string escapeJson(const std::string& json) const {
+    rapidjson::StringBuffer buffer;
+    rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+    writer.String(json.c_str(), json.size());
+    return buffer.GetString();
+  }
+
+  std::vector<std::string> getOutputFileContents() {
+    std::vector<std::string> file_contents;
+
+    auto callback = [&file_contents](const std::string& path, const std::string& filename) -> bool {
+      std::ifstream is(path + utils::file::FileUtils::get_separator() + filename, std::ifstream::binary);
+      std::string file_content((std::istreambuf_iterator<char>(is)), std::istreambuf_iterator<char>());
```

Comment:
It's needed because of a vexing-parse warning where the `std::istreambuf_iterator<char>(is)` could be interpreted as a parameter declaration. Here is an explanation of this: https://stackoverflow.com/questions/7007817/a-confusing-detail-about-the-most-vexing-parse

Reply:
OK, thanks.  I would probably use `{}` instead of the extra `()`:
```c++
       std::string file_content(std::istreambuf_iterator<char>{is}, std::istreambuf_iterator<char>{});
```
but it's fine as it is, too.

Reply:
Thanks it looks better this way, updated in 4e8f82bc2e1f7997a7d51dab9ff413fc0a1d0e13